simpleviewinc/keg-cli
# Kill by port

## Context

* I often want to know which processes are using a specific port
* I often want to kill these, since they may be processes that I tried to close but somehow they still linger around
* the bash commands to do this are annoying and easy to forget

## Goal

* Create some keg-cli tasks for listing and killing processes using a port

## Updates
- `src/tasks/network/killProcessUsingPort.js`
  * the task for killing a process using a port
- `src/tasks/network/listProcessesUsingPort.js`
  * the task for listing processes using a port in a table
- `src/utils/network/getProcessesUsingPort.js`
  * the util function for getting all processes using a port
- `src/utils/network/printProcesses.js`
  * the util function for printing those processes in a table
- `src/utils/process/kill.js`
  * the util function for killing a process by its pid

## Testing
* start up any dev server or process that uses a port
* `keg net list <port>`
* verify that the process you started appears in the table
* `keg net kill <port>`
* verify that the cli prints out the same list, and asks you to confirm to kill the processes. Type n/no.
* verify the process still runs
* Run it again, this time enter y/yes
* verify the process was killed
* start the same process up again
* `keg net kill --port <port> --force`
* verify that it killed the process without prompting you to confirm first
